app/src/main/java/org/mozilla/focus/searchsuggestions/SearchSuggestionsFetcher.kt

```diff
@@ -0,0 +1,78 @@
+/* This Source Code Form is subject to the terms of the Mozilla Public
+ * License, v. 2.0. If a copy of the MPL was not distributed with this
+ * file, You can obtain one at http://mozilla.org/MPL/2.0/. */
+
+package org.mozilla.focus.searchsuggestions
+
+import android.arch.lifecycle.LiveData
+import android.arch.lifecycle.MutableLiveData
+import kotlinx.coroutines.experimental.CommonPool
+import kotlinx.coroutines.experimental.android.UI
+import kotlinx.coroutines.experimental.channels.Channel
+import kotlinx.coroutines.experimental.channels.consumeEach
+import mozilla.components.browser.search.suggestions.SearchSuggestionClient
+import kotlinx.coroutines.experimental.launch
+import mozilla.components.browser.search.SearchEngine
+import okhttp3.OkHttpClient
+import okhttp3.Request
+import org.mozilla.focus.utils.debounce
+
+open class SearchSuggestionsFetcher(searchEngine: SearchEngine) {
+    data class SuggestionResult(val query: String, val suggestions: List<String>)
+
+    private var client: SearchSuggestionClient? = null
+    private var httpClient = OkHttpClient()
+
+    private val fetchChannel = Channel<String>(capacity = Channel.UNLIMITED)
+
+    private val _results = MutableLiveData<SuggestionResult>()
+    val results: LiveData<SuggestionResult> = _results
+
+    var canProvideSearchSuggestions = false
+        private set
+
+    init {
+        updateSearchEngine(searchEngine)
+        launch(CommonPool) {
+            fetchChannel
+                    .debounce(THROTTLE_AMOUNT)
+                    .consumeEach { getSuggestions(it) }
```

Comment:
Good question. I think it will always send the last one at the end of the `onTimeout`

Also In the `fetch` I cancel any other jobs that are in flight